core/App/ehsm_provider.cpp

```diff
@@ -835,4 +839,127 @@ ehsm_status_t generate_apikey(ehsm_data_t *apikey)
     }
 }
 
+ehsm_status_t ra_get_msg1(sgx_ra_msg1_t *msg1)
+{
+    sgx_status_t ret = SGX_ERROR_UNEXPECTED;
+
+    if (msg1 == NULL) {
+        return EH_ARGUMENTS_BAD;
+    }
+
+    //initialize the ra session
+    sgx_status_t sgxStatus = SGX_SUCCESS;
+    int enclave_lost_retry_time = 1;
+    do {
+        ret = enclave_init_ra(g_enclave_id, &sgxStatus, false, &g_context);
+        //Ideally, this check would be around the full attestation flow.
+    } while (SGX_ERROR_ENCLAVE_LOST == ret && enclave_lost_retry_time--);
+
+    if(SGX_SUCCESS != ret || sgxStatus){
+        printf("Error, call enclave_init_ra fail [%s].\n", __FUNCTION__);
+       return EH_FUNCTION_FAILED;
+    }
+    printf("Call enclave_init_ra success.\n");
+
+    //get the msg1 from core-enclave 
+    sgx_att_key_id_t selected_key_id = {0};
+    ret = sgx_ra_get_msg1_ex(&selected_key_id, g_context, g_enclave_id, sgx_ra_get_ga, msg1);
+    if(SGX_SUCCESS != ret) {
+        printf("Error, call sgx_ra_get_msg1_ex failed(%#x)\n", ret);
+        return EH_FUNCTION_FAILED;
+    }
+    return EH_OK;
+}
+
+ehsm_status_t ra_get_msg3_quote_size(uint32_t *quote_size)
+{
```

Comment:
is need?

Reply:
can the napi directly call the funtion sgx_qe_get_quote_size?

Reply:
It will be fix